cupy/cuda/compiler.py

```diff
@@ -103,7 +102,8 @@ def get_cache_dir():
 _empty_file_preprocess_cache = {}
 
 
-def compile_with_cache(source, options=(), arch=None, cache_dir=None):
+def compile_with_cache(source, options=(), arch=None, cache_dir=None,
+                       extra_source=None):
```

Comment:
Write a comment why `extra_source` is required. It is because we cannot use nvrtc to get result of preprocess though want to check if header files are changed.